src/server/pkg/hashtree/cache.go

```diff
@@ -4,35 +4,81 @@ import (
 	"fmt"
 	"io"
 	"reflect"
+	"sync"
 
-	lru "github.com/hashicorp/golang-lru"
+	"github.com/hashicorp/golang-lru/simplelru"
 	"github.com/pachyderm/pachyderm/src/server/pkg/localcache"
 	"github.com/sirupsen/logrus"
 )
 
 // Cache is an LRU cache for hashtrees.
 type Cache struct {
-	*lru.Cache
+	lruCache  simplelru.LRUCache
+	lock      sync.Mutex
+	syncEvict *bool
```

Comment:
Added this `syncEvict` pointer so that we can make evictions synchronous while tearing down the cache.